addons/web_editor/static/lib/odoo-editor/src/utils/sanitize.js

```diff
@@ -192,16 +191,17 @@ class Sanitize {
             ) {
                 node.setAttribute('contenteditable', 'false');
             }
+
             if (node.firstChild) {
                 this._parse(node.firstChild);
             }
+
             // Update link URL if label is a new valid link.
             if (node.nodeName === 'A' && anchorEl === node) {
-                const linkLabel = node.textContent;
-                const match = linkLabel.match(URL_REGEX);
-                if (match && match[0] === node.textContent && !node.href.startsWith('mailto:')) {
-                    const urlInfo = getUrlsInfosInString(linkLabel)[0];
-                    node.setAttribute('href', urlInfo.url);
+                const linkLabel = node.innerText;
+                const urlInfo = getUrlsInfosInString(linkLabel);
+                if (urlInfo.length && urlInfo[0].label === linkLabel && !node.href.startsWith('mailto:')) {
```

Comment:
When rebasing to force-push, I had to incorporate the changes introduced by this commit:
https://github.com/odoo/odoo/pull/107603/commits/1d8ca674c31ddd60680f889cf25d6c52740b54a0

Unless I'm missing something here, the commit does not do what it says it does: it does not "set link label as new URL". It simply prevents the href update mechanism.
It does solve the problem of breaking mailto links with the addition of "https://", but does not update the link if the email address is changed.

Such suitable link update (working for both https and mailto links) is the subject of task https://www.odoo.com/web#id=3102761&menu_id=4720&cids=1&action=333&active_id=1695&model=project.task&view_type=form

Should I take on this task in this PR? Or should I do it on a new PR targeting master?

Reply:
You don't _have_ to do it in this PR but if you think it would be easier to do so (I suspect it would in terms of future conflicts) then of course you can do it ! :)

Reply:
A proper link update supporting `http`, `mailto` and `tel` links is being done at https://github.com/odoo/odoo/pull/111521 targeting master. I'd rather leave this PR as it is, and later back-port the link update part for `mailto` if necessary.